docs/feature-requests/006-pull-request-reviewer-flow.md

```diff
@@ -0,0 +1,156 @@
+
+User can respond to a comment thread by adding a single line comment (current implementation) or starting a new review. The two buttons should only show up when the comment textbox is in focus, or is _not_ empty.
+
+When there is already an existing pending review, there should only be **one** `btn-primary` button that reads "Comment".
+
+#### Pending comments
+
+![pending comment](https://user-images.githubusercontent.com/6842965/56692893-2ce59a00-66b1-11e9-81cc-bc7956bc8bec.png)
+
+Pending comments within the All Reviews tab are styled differently from the already published comments. Pending comments contain a badge, and when clicked, will take user to the Pending Review tab.
+
+
+### "Pending Review" tab
+This tab shows *a subset* of all reviews -- only the summary and comments of a pending review. Since a user is only allowed to have one pending review at a time, there should also only be maximum one Pending Review tab.
+
+#### Header (or the alternate footer)
+The header looks very similar to the one of All Reviews tab, with the exception that the primary button now reads "See all reviews", and will send users back to the All Reviews tab.
+
+
+#### Summary section
+
+![pending review summary](https://user-images.githubusercontent.com/6842965/56699584-23196200-66c4-11e9-94a4-193c9d662bb3.png)
+
+The summary section of the Pending Review tab is sticky (although still collapsible), so it stays within view regardless of how long the list of comments below it is. The icon on the left indicates the type of review, which can be selected in the dropdown underneath the text box. The button to submit review will be disabled a review type has not been chosen from the dropdown menu.
+
+
+#### Comments section
+
+![pending review comments](https://user-images.githubusercontent.com/6842965/56699828-31b44900-66c5-11e9-948c-a5c03215e5d8.png)
+
+The comments section of the Pending Review tab looks very similar to that of the All Reviews tab, except that the progress bar is replaced by a small comment counter on top of the whole section.
+
+**Empty State** of this section should contain a picture tutorial of how to add a comment via gutter icon.
+
+
+### New Comment
+![new comment](https://user-images.githubusercontent.com/6842965/56695406-fdd22700-66b6-11e9-9e7e-fe85e2507a66.png)
+
+A new comment block can appear in either All Reviews tab or Pending Review tab, depending on the scenarios covered in [the section below](#add-comment-gutter-icon). When in focus, a new comment block always has a glowing border to emphasize itself. If there is already a pending review, there should only be one `btn-primary` button that reads "Comment".
+
+
+### "Add comment" gutter icon
+
+A user can start a review or add a comment to an existing pending review by clicking on the "add comment" icon which shows up on hover over the gutter of:
+
+1. MultiFilePatch view within Files tab in `PullRequestDetailView`
+2. an editor *if on a checked out PR branch*
```

Comment:
The big question left for me: we can only create comments in regions _modified by the pull request_. How do we want to signal which areas are able to be commented upon? Can we give users a quick way to navigate from one to another, so they don't get "lost" trying to find another place to comment on?

Reply:
Yeah, I was wondering the same thing. 

> we can only create comments in regions modified by the pull request

... plus a few adjacent context lines.

While this would be pretty cool, my gut is to first ship with comment creation on the `PullRequestDetailView` and leave in-editor comment creation out of scope since it has these extra complexities. 

If you think about the reviewer flow, the diff is usually what's being looked at when there is a need to create a comment. Jumping to the file could be helpful for extra context, but I think it's unlikely that a user will be in an editor first and realize they want to make a comment. And we can always try adding this in after the initial ship.

Reply:
Yup, definitely didn't realize you can only comment on the modified regions! That changed things considerably (in a good way) -- think it makes sense to scope it down to only allowing comments to be made on `PullRequestDetailView` like @kuychacho suggested, at least for the first iteration.